src/rdkafka_admin.h

```diff
@@ -476,6 +476,186 @@ struct rd_kafka_ConsumerGroupDescription_s {
         /** Group specific error. */
         rd_kafka_error_t *error;
 };
+/* MY STRUCTS START!!*/
+enum rd_kafka_ScramMechanism_s {
+        UNKNOWN = 0,
+        SCRAM_SHA_256 = 1,
+        SCRAM_SHA_512 = 2
+};
+struct rd_kafka_ScramCredentialInfo_s {
+        rd_kafka_ScramMechanism_t mechanism;
+        int32_t iterations; 
+}
+void rd_kafka_ScramCredentialInfo_set_mechanism(rd_kafka_ScramCredentialInfo_t *scram_credential_info,rd_kafka_ScramMechanism_t mechanism){
+        scram_credential_info->mechanism = mechanism;
+}
+void rd_kafka_ScramCredentialInfo_set_iterations(rd_kafka_ScramCrdentialInfo_t *scram_credential_info,int32_t iterations){
+        scram_credential_info->iterations = iterations;
+}
+rd_kafka_ScramMechanism_t rd_kafka_ScramCredentialInfo_get_mechanism(rd_kafka_ScramCredentialInfo_t *scram_credential_info){
+        return scram_credential_info->mechanism;
+}
+int32_t rd_kafka_ScramCredentialInfo_get_iterations(rd_kafka_ScramCrdentialInfo_t *scram_credential_info){
+        return scram_credential_info->iterations;
+}
+struct rd_kafka_UserScramCredentialsDescription_s {
+        char *user;
+        rd_kafka_error_t *error;
+        size_t credential_info_cnt;
+        rd_kafka_ScramCredentialInfo_t *credential_infos;
+}
+rd_kafka_UserScramCredentialsDescription_t *rd_kafka_UserScramCredentialsDescription_new(size_t num_credentials){
+        rd_kafka_UserScramCredentialsDescription_t *description;
+        description = rd_calloc(1,sizeof(*description));
+        description->credential_info_cnt = num_credentials;
+        rd_kafka_ScramCredentialInfo_t *credentialinfo;
+        description->credential_infos = rd_calloc(num_credentials,sizeof(*credentialinfo));
```

Comment:
Question: is it for sure that num_credentials > 0 always? Otherwise rd_calloc can fail. Just something to keep in mind.

Reply:
Will take care of it !